### DISCUSSION TITLE
Web Workers and synchronous calls

### DISCUSSION THREAD

**Discussion by bergwerf** [rdkit/rdkit-js#450] [General]
I noticed that this library uses WebAssembly, and that the initialization is async. However, subsequent calls to the library are sync (e.g. they do not have to be awaited). I imagined that all work would be done in a Web Worker running the compiled WASM, but then all calls should have been async. How does this work exactly?

For my application I am interested in using the CoordGen 2D layout algorithm. It might be valuable if this computation happens async, such that it doesn't block the UI thread when it isn't instant.

**Comment by ptosco** [comment]
@bergwerf I suggest you take a look at https://github.com/rdkit/rdkit-structure-renderer, which is also available as `npm` package. It indeed does rendering using Web Workers to avoid blocking the UI thread. You may configure the number of Web Workers you wish to start. Have a look at the example `index.html` page and at the `api_examples` for inspiration.
